exasol/ds/sandbox/runtime/ansible/roles/jupyter/files/notebook/script_languages_container/using_a_script_languages_container.ipynb

```diff
@@ -0,0 +1,704 @@
+   "metadata": {},
+   "source": [
+    "## Customizing Script-Language Containers\n",
+    "Sometimes you need very specific dependencies or versions of dependencies in the Exasol UDFs. In such case you can customize a Script-Language Container.\n",
+    "### Flavor Definition\n",
+    "![image.png](attachment:47f643bc-704e-4b00-8405-917b5fdae990.png)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "58895979-99f7-4fad-ab0f-9cfa020b3750",
+   "metadata": {},
+   "source": [
+    "For customizing a flavor usually the flavor_customization build step is most important. It contains everything you need to add dependencies. The remaining build steps should be only changed with care, but sometimes some dependencies are defined in other build steps because the script client depends on them. \n",
+    "Let's check the directory structure of the selected flavor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "11f3d8ae-d23b-4968-a5c3-a18de8f434d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "show_directory_content(slctmanager.slc_dir.flavor_dir, 2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "67752053-c6a6-476f-892f-613113c1668b",
+   "metadata": {},
+   "source": [
+    "### Flavor Customization Build Step"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fbfdb5b8-4c9f-4174-9752-8652013630e4",
+   "metadata": {},
+   "source": [
+    "The **flavor_customization** build step consists of a Dockerfile and several package lists which can be modified. We recommend to use the package lists to add new packages to the flavor and only modify the Dockerfile if you need very specific changes, like adding additional resources."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "de268b34-99e3-4af6-a9b3-02a44e803c9f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "show_directory_content(slctmanager.slc_dir.flavor_dir / \"flavor_customization\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5803c74e-2f7f-4d77-b363-f5968d1937bc",
+   "metadata": {},
+   "source": [
+    "The Dockerfile consists of two parts. The first part installs the packages from the package list and should only be change with care. The second part is free for your changes. Read the description in the Dockerfile carefully to find out what you can and shouldn't do."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "562eac65-b205-49d4-971e-9ed0090796b4",
+   "metadata": {},
+   "source": [
+    "#### Package Lists\n",
+    "The package lists have a unified format. Each line consists of the package name and the package version separated by \"|\", e.g `xgboost|1.3.3`. You can comment out a whole line by adding\"#\" at the beginning. You can also add a trailing comment to a package definition by adding a \"#\" after the package definition. We usually recommend to pin the version, such that there are no surprises for which version gets installed."
```

Comment:
```suggestion
    "The package lists have a unified format. Each line consists of the package name and the package version separated by a pipe-symbol `|`, e.g `xgboost|1.3.3`. You can comment out a whole line by prefixing it with a hash mark `#`. You can also add a trailing comment to a package definition by adding a `#` after the package definition. We usually recommend to pin the version, in order to avoid surprises about the actually installed version."
```